src/main/splitting/task_checker.py

```diff
@@ -0,0 +1,144 @@
+
+
+# returns True if time is out, because it means that no other errors were raised,
+# so in case of code correctness checking it means that code is correct
+def check_call_safely(call_args: list, timeout=consts.SUBPROCESS_TIMEOUT, timeout_return=True):
+    try:
+        check_call(call_args, timeout=timeout)
+        return True
+    except CalledProcessError as e:
+        log.exception(e)
+        return False
+    except TimeoutExpired as e:
+        log.exception(e)
+        return timeout_return
+
+
+def remove_compiled_files():
+    remove_directory(SOURCE_FOLDER)
+    create_directory(SOURCE_FOLDER)
+
+
+class ITaskChecker(object, metaclass=ABCMeta):
+    @property
+    @abstractmethod
+    def language(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def min_symbols_number(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def output_strings(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def create_source_file(self, source_code: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_source_file_correct(self, source_file: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def run_test(self, input: str, expected_output: str, source_file: str):
+        raise NotImplementedError
+
+    @staticmethod
+    def get_no_need_to_run_tests_values(rate: float, tasks_len: int):
+        need_to_run_tests = False
+        test_results = [rate] * tasks_len
+        return need_to_run_tests, test_results
+
+    def check_before_tests(self, source_file: str, source_code: str, tasks: list):
+        test_results = []
+        need_to_run_tests = True
+        rate = consts.TEST_RESULT.CORRECT_CODE.value
+
+        # not to check incorrect fragments
+        if not self.is_source_file_correct(source_file):
+            rate = consts.TEST_RESULT.INCORRECT_CODE.value
+            need_to_run_tests, test_results = self.get_no_need_to_run_tests_values(rate, len(tasks))
+
+        # not to check too small fragments because they cannot return true anyway
+        elif len(source_code) < self.min_symbols_number:
+            log.info('Code fragment is too small')
+            need_to_run_tests, test_results = self.get_no_need_to_run_tests_values(rate, len(tasks))
+
+        # not to check fragments without output because they cannot return anything
+        elif not does_string_contain_any_of_substrings(source_code, self.output_strings):
+            log.info('Code fragment does not contain any output strings')
+            need_to_run_tests, test_results = self.get_no_need_to_run_tests_values(rate, len(tasks))
+
+        return need_to_run_tests, test_results, rate
+
+    def check_task(self, task: str, in_and_out_files_dict: dict, source_file: str, stop_after_first_false=True):
+        log.info(f'Start checking task {task}')
+        in_and_out_files = in_and_out_files_dict.get(task)
+        if in_and_out_files is None:
```

Comment:
I think we should check if len(in_and_out_files) is zero, and if it is, we should raise an error because it is our mistake (we forgot about in and out files for the task and then we will have the division by zero error during rate calculating further). Maybe it is better the check of it to put in the task_tests_handler file because it is an init state